File: pantry-tracker-backend/database_operations/getGroceryList.py
from flask import Blueprint, jsonify
from sqlalchemy import text
from extensions import db

getGroceryList_bp = Blueprint('getGroceryList', __name__)
@getGroceryList_bp.route('/get-grocery-list', methods=['GET'])

def get_grocery_list():
    try:
        result = db.session.execute(text('SELECT gl.aliased_ingredient_name, gl.quantity, u.unit_name FROM my_grocery_list gl JOIN units u ON gl.unit_id = u.id WHERE gl.quantity > 0 ORDER BY gl.aliased_ingredient_name'))
        ingredients = [{'name': row[0], 'quantity': row[1], 'unitID': row[2]} for row in result]
        return jsonify({'ingredients': ingredients})
    except Exception as e:
        return jsonify({'error': str(e)}), 500

File: pantry-tracker-frontend/src/supabase_db_operations/getGroceryList.js
import supabaseClient from '../utils/SupabaseClient';

const getGroceryList = async (userID) => {
    try {
        const { data: userIngredients, error } = await supabaseClient
            .from('user_ingredients')
            .select(`
                aliased_ingredient_name,
                gl_quantity,
                units:gl_unit_id(unit_name)
            `)
            .eq('user_id', userID)
            .gt('gl_quantity', 0)
            .order('aliased_ingredient_name', { ascending: true});

        if (error) {
            throw (error);
        };

        console.log('ingredients', userIngredients);
        const ingredients = userIngredients.map(ingredient => ({
            name: ingredient.aliased_ingredient_name,
            quantity: ingredient.gl_quantity,
            unitID: ingredient.units.unit_name
        }));
        
        return ingredients;

    }
    catch (error) {
        console.error(error);
        return { 'error': error.message };
    };
};

export default getGroceryList;